### PR TITLE
Treat missing depfile as a empty list of dependencies.

### LINKED ISSUE #80: Tries to open nonexistent .d file in LLVM ASM build
Hi,

Another weird one (sorry!). After #78 was fixed I was able to build .cpp files from the LLVM repo on Windows, but I run into problems with ASM files, here is the error:

```
read lib\Support\BLAKE3\CMakeFiles\LLVMSupportBlake3.dir\blake3_avx2_x86-64_windows_msvc.asm.obj.d: The system cannot find the file specified. (os error 2)
```

Here is the command invocation:

```
C:\PROGRA~1\MIB055~1\2022\PROFES~1\VC\Tools\MSVC\1436~1.325\bin\Hostx64\x64\ml64.exe -DGTEST_HAS_RTTI=0 -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_HAS_EXCEPTIONS=0 -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -IC:\code\llvm\llvm-project\out\debug\lib\Support\BLAKE3 -IC:\code\llvm\llvm-project\llvm\lib\Support\BLAKE3 -IC:\code\llvm\llvm-project\out\debug\include -IC:\code\llvm\llvm-project\llvm\include  -c -Fo lib\Support\BLAKE3\CMakeFiles\LLVMSupportBlake3.dir\blake3_avx2_x86-64_windows_msvc.asm.obj C:\code\llvm\llvm-project\llvm\lib\Support\BLAKE3\blake3_avx2_x86-64_windows_msvc.asm
failed: Building ASM_MASM object lib\Support\BLAKE3\CMakeFiles\LLVMSupportBlake3.dir\blake3_avx2_x86-64_windows_msvc.asm.obj
```

I am *guessing* that n2 tries to find the dependency file for the .asm file, but ml64 doesn't output (maybe no assemblers do?). Hopefully an easy fix.

### PR DESCRIPTION
**Pull request by tru**
CMake can sometimes write depfile directives to build.ninja even when no depfile is generated. This was handled by "ninja" by just ignoring the missing file and going on with the build, see the code here:

https://github.com/ninja-build/ninja/blob/master/src/build.cc#L894

This fixes #80

### CODE REVIEW

**Comment by Colecf** [comment]
It might be best to guard this behavior behind `fake_ninja_compat`, the error does seem good to have in the general case. Ideally the bug in cmake would be fixed so it doesn't try to use nonexistent depfiles.

(But I'm not a maintainer, I'll leave it up to Evan to decide)

**Comment by evmar** [comment]
I traced the origin of this behavior to this change, hrm

https://github.com/ninja-build/ninja/commit/c3c1b3fda8b415e917b2762526c0c86e099b4300

File: src/task.rs [hunk 1]
```diff
@@ -40,8 +40,15 @@ pub struct TaskResult {
 fn read_depfile(path: &Path) -> anyhow::Result<Vec<String>> {
     let mut bytes = match std::fs::read(path) {
         Ok(b) => b,
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Vec::new(),
```
Comment (evmar) [inline src/task.rs:43]: I think this would be better if this returned here, so we don't need to use bytes.len() as a marker for missing file.

**Comment by evmar** [comment]
One thing I'm especially worried about here is whether this is actually a bug in how n2 interprets the CMake output.  I poked a little bit on #80 just to double check.

**Comment by evmar** [comment]
I wrote some notes on #80, filed #99 about the larger validation problem, and merged this with small changes.  Thanks for writing the patch!

**Comment by Colecf** [comment]
It turns out android already makes this an error: https://cs.android.com/android/platform/superproject/main/+/main:build/soong/ui/build/ninja.go;l=78;drc=ded796030ba24c6ea096c26067890884825f9ebb

So we'd need the option to turn it into an error again if we were to switch to n2.

**Pull request closed without merging**